### PR TITLE
metadata endpt changes

### PR DESCRIPTION
**Pull request by medullaskyline**
Haven't tested yet (VirtualBox issues).

### CODE REVIEW

**Comment by phyllers** [comment]
Has this been tested yet?

**Comment by medullaskyline** [comment]
no

File: api/metadata.py [hunk 1]
```diff
@@ -2075,12 +2081,10 @@ def metadata_attr_list(self, request):
 
             cursor.close()
             db.close()
-
+            request_finished.send(self)
             return MetadataAttrList(items=data, count=len(data))
 
         except (IndexError, TypeError):
-            if cursor: cursor.close()
-            if db: db.close()
```
Comment (s-paquette) [inline api/metadata.py]: I don't see this db.close in the finally; it should be added there to make sure the connection is closed in the case of an exception, like the cursor line is. (Unless that's not needed in Python?)

File: api/metadata.py [hunk 2]
```diff
@@ -2075,12 +2081,10 @@ def metadata_attr_list(self, request):
...
             raise endpoints.InternalServerErrorException('Error retrieving attribute list')
         finally:
             if cursor: cursor.close()
```
Comment (medullaskyline) [inline api/metadata.py]: `if db: db.close()` is called in the `finally` clause in  line 2087. Am I misunderstanding your line note, @s-paquette?

Actually now that I'm looking at it it should be `if db and db.open: db.close()`.

Comment (s-paquette) [inline api/metadata.py]: Oh, maybe it's not showing--here in the diffs all I can see is the db.close being removed from except and not added to the finally, but maybe that's just github being github-y (or it was already there).

**Pull request merged**